flink-runtime/src/main/scala/org/apache/flink/runtime/taskmanager/TaskManager.scala

```diff
@@ -781,6 +787,32 @@ class TaskManager(
     }
   }
 
+  private def handleRequestTaskManagerLog
+  (sender: ActorRef, requestType: LogTypeRequest, jobManager: ActorRef):
+  Unit = {
+    if (blobService.isDefined && jobManager.path.address.host.isDefined) {
+      //create new Thread to upload log, to not block the TM for too long
+      new Thread() {
+        override def run(): Unit = {
+          val jmHost = jobManager.path.address.host.get
+          val port = blobService.get.getPort
+          val logFilePath = System.getProperty("log.file");
+          val file: File = requestType match {
+            case LogFileRequest => new File(logFilePath);
+            case StdOutFileRequest =>
+              new File(logFilePath.substring(0, logFilePath.length - 4) + ".out");
+          }
+
+          val client: BlobClient = new BlobClient(new InetSocketAddress(jmHost, port))
+
+          sender ! client.put(new FileInputStream(file))
+        }
+      }.run()
```

Comment:
This won't start a new thread but simply call the just overwritten `run` method. Use `start` instead. Furthermore, it is not encouraged to extend the Thread class. It is better to implement a `Runnable` or even better avoiding to deal with raw threads at all. I guess it would be easier to simply spawn a `Future` which is executed by a respective `ExecutionContext`.